oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -573,54 +575,53 @@ private List<Query> nonFullTextConstraints(IndexPlan plan, PlanResult planResult
 
         Filter filter = plan.getFilter();
         if (!filter.matchesAllTypes()) {
-            queries.add(nodeTypeConstraints(planResult.indexingRule, filter));
+            nodeTypeConstraints(planResult.indexingRule, filter).ifPresent(queries::add);
+            //queries.add(nodeTypeConstraints(planResult.indexingRule, filter));
```

Comment:
The comment was a leftover.

I personally like concise code but I see it could be less readable. I have made it a little more verbose. Hope is a good compromise.